mesh/mesh.cpp

```diff
@@ -3508,6 +3506,8 @@ void Mesh::SetMeshGen()
             mesh_geoms |= (1 << Geometry::SQUARE);
             mesh_geoms |= (1 << Geometry::SEGMENT);
             mesh_geoms |= (1 << Geometry::POINT);
+         case Element::SEGMENT:
+            mesh_geoms |= (1 << Geometry::SEGMENT);
```

Comment:
I am slightly concerned about this change. It's clear that SEGMENT can be regarded as both "simplex" and "tensor product", depending on context. I am not 100% sure this won't break something else, but let's have this tested...

Reply:
Yes, I agree. I am not sure I fully understand the consequences or intent of the meshgen variable. Testing should tell us something.

Reply:
Yes, I agree. I am not sure I fully understand the consequences or intent of the meshgen variable. Testing should tell us something about it anyway.

I guess another avenue here would be to modify the conditional I pointed to above. But I'm not entirely sure what modification should be made.

Reply:
I guess I didn't prefer that option because I though it should be default to treat 1D meshes this way, since it has more functionality. The other reason was that it seemed non-intuitive to be setting a flag with that name for a 1D mesh. So I wanted to pick up the behavior by default.
But certainly that would eliminate these other issues.

Reply:
I like Veselin's suggestion, if that solves the problem it's probably better to keep the SEGMENT meshgen as it is...